snuba/query/query.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from deprecation import deprecated
```

Comment:
It is explicitly imported in setup.py
> setup.py:        'deprecation==2.0.3',
Though it was not used.

I think the annotation has a value: whoever adds more code that is using the query class would think twice (hopefully) before using a method that is marked as deprecated and that provides guidance on what to use instead. It also points out that the method is there because of a migration and that is not part of the intended design.
I agree that we should keep on top of deprecations.

Reply:
Oops — I think I wrote this with the intention to double check that this was accurate and forgot. Sorry!